src/components/structures/MatrixChat.js

```diff
@@ -1022,17 +1029,18 @@ module.exports = React.createClass({
                     page_element = (
                         <RoomView
                             ref="roomView"
-                            roomAddress={this.state.currentRoom || this.state.currentRoomAlias}
+                            roomAddress={this.state.currentRoomAlias || this.state.currentRoomId}
```

Comment:
(maybe the easiest fix for all this would actually be to make peeking-by-alias work in synapse...)